PiwikTracker/Device.swift

```diff
@@ -0,0 +1,132 @@
+import UIKit
+
+internal struct Device {
+    
+    /// A shared instance holding the properties for the current device
+    static let current:Device = Device(platform: currentPlatform(),
+                                       humanReadablePlatformName: currentHumanReadablePlatformName(),
+                                       osVersion: currentOSVersion(),
+                                       screenSize: currentScreenSize(),
+                                       nativeScreenSize: currentNativeScreenSize())
+    /// The platform name of the device i.e. "iPhone1,1" or "iPad3,6"
+    let platform: String
+    
+    /// A human readable version of the platform name i.e. "iPhone 6 Plus" or "iPad Air 2 (WiFi)"
+    /// Or the platform name in case no human readable name was found.
+    let humanReadablePlatformName: String
```

Comment:
Agree that returning an optional String is probably a good thing.
If the caller really needs a string it can do 
````
let platformName = device. humanReadablePlatformName ?? device = platform 
````
But that is the callers responsibility not this struct.